packages/core/src/connectivity/scp-cf/token-accessor.ts

```diff
@@ -40,40 +69,48 @@ export async function serviceToken(
       userJwt?: string | JwtPayload;
     }
 ): Promise<string> {
-  const resolvedService = resolveService(service);
-
   const opts = {
     useCache: true,
     enableCircuitBreaker: true,
     ...options
   };
 
+  service = resolveService(service);
+  const serviceCredentials = service.credentials;
   const xsuaa = multiTenantXsuaaCredentials(opts.userJwt);
-  const serviceCreds = extractClientCredentials(resolvedService.credentials);
 
   if (opts.useCache) {
     const cachedToken = clientCredentialsTokenCache.getGrantTokenFromCache(
       xsuaa.url,
-      serviceCreds
+      serviceCredentials.clientid
     );
     if (cachedToken) {
       return cachedToken.access_token;
     }
   }
+
+  // TODO: handle Payload as well
+  let userJwt = options?.userJwt;
+  if (typeof userJwt !== 'string' && typeof userJwt !== 'undefined') {
+    userJwt = undefined;
+  }
+
   try {
-    const resp = await clientCredentialsGrant(xsuaa, serviceCreds, opts);
+    const token = await getClientCredentialsToken(service, userJwt);
+
     if (opts.useCache) {
       clientCredentialsTokenCache.cacheRetrievedToken(
         xsuaa.url,
-        serviceCreds,
-        resp
+        serviceCredentials.clientid,
+        token
       );
     }
-    return resp.access_token;
-  } catch (error) {
+    // TODO: Token responses can have id_token instead of access_token (xssec does that correctly)
```

Comment:
also here?

Reply:
I am aware of this one, but this is the old behavior therefore I wouldn't touch it on this PR either. I'll add it to the ACs of the ticket to consider/discuss this.

Reply:
Perhaps you can explain what is the difference between the two and will there be a separate property? In the type I only see `access_token`